storage/replica.go

```diff
@@ -492,6 +492,11 @@ func (r *Replica) setReplicaID(replicaID roachpb.ReplicaID) error {
 
 // setReplicaIDLocked requires that the replica lock is held.
 func (r *Replica) setReplicaIDLocked(replicaID roachpb.ReplicaID) error {
+	if replicaID == 0 {
```

Comment:
This changes behavior. Take the following example:
- replicate to three nodes
- take a node down
- down replicate that same node; downed node won't get the memo
- bring down node back up
- up-replicate again to formerly down node
- pre-emptive snapshot will be sent with `replicaID=0`; previously, this would yield an error here "cannot move backwards from X to 0". Since now we don't return an error, it means we'll be applying the snapshot.

Not necessarily a bad thing, though it's different behavior and I'm not familiar enough with the code to know whether overwriting will potentially cause issues.